javascript/node/selenium-webdriver/test/testing/index_test.js

```diff
@@ -105,3 +105,59 @@ describe('Mocha Integration', function() {
     });
   });
 });
+
+describe('Mocha async "done" support', function() {
+   this.timeout(2*1000);
+
+   var waited = false;
+   var DELAY = 100; // ms enough to notice
+
+   // --- First, vanilla mocha "it" should support the "done" callback correctly.
+
+   // This it blocks until 'done' is invoked
+   it('vanilla delayed', function it_vanillaDelayed(done) {
+      setTimeout(function delayed_vanillaTimeout() {
+         waited = true;
+         done();
+      }, DELAY);
+   });
+
+   it('vanilla must have waited', function it_vanillaMustHaveWaited() {
+      assert.strictEqual(waited, true);
```

Comment:
Tests should not depend on ordering.  Use

    beforeEach(function() { waited = false; });
    afterEach(function() { assert.strictEqual(waited, true); });